### ISSUE TITLE
Issue with Long-Range StormCast UNet Simulations

### ISSUE THREAD

**Issue by Steven04453X** [NVIDIA/physicsnemo#1351]
Dear StormCast Team,
I hope this message finds you well. I am writing to ask for your advice regarding an issue I encountered while running extended forecast with the StormCast UNet model.

I trained the StormCast UNet model using HRRR and ERA5 datasets from 2018 to 2024, except for 2022, which was reserved for validation. Figure 1 shows the evolution of the loss function during training.
 

<img width="379" height="343" alt="Image" src="https://github.com/user-attachments/assets/66d5f64a-dcac-481f-b1a8-ad9bd96c757b" />

Figure 1. Loss curves during weighted training. The training dataset contains 52800 samples, while the validation dataset contains 8800 samples. MA denotes the moving average.

When conducting forecasts longer than four days, however, unexpected behavior emerges. As an example, I examined an atmospheric river event from Mar 8th to 20th, 2017. The StormCast UNet model was initialized with HRRR data at 00:00 UTC on Mar 8th, 2017, and the ERA5 fields were used as synoptic-scale forcing, updated hourly. The simulation was run for 10 days (Mar 8th to 18th, 2017). 

Figure 2 presents daily-mean fields of specific humidity at 850 hPa (A, D, G, J), wind speed at 850 hPa (B, E, H, K), and precipitation (C, F, I, L). The model produces reasonable simulations during the first 96 lead hours, but the forecast degrades substantially beyond that point.
 

<img width="334" height="456" alt="Image" src="https://github.com/user-attachments/assets/c203c856-8147-4a9a-896b-076ed39ec24a" />

Figure 2. Daily-mean specific humidity at 850 hPa (A, D, G, J), wind speed at 850 hPa (B, E, H, K), and precipitation (C, F, I, L) for the first (A to C), second (D to F), fourth (G to I) and fifth (J to L) forecast days.

To further diagnose the issue, Figure 3 shows the time series of daily total precipitation averaged over the Western US (30ºN~50ºN, 115ºW~130ºW). While precipitation during the first three days remains within a reasonable range, it increases unrealistically beginning around Mar 10th, 2017.

We also tested a nudging configuration in which the mesoscale state near the domain boundaries is replaced with HRRR data. However, similar issues persist in the nudged simulations.
 

<img width="468" height="335" alt="Image" src="https://github.com/user-attachments/assets/d1c73fbf-4493-4b42-84b9-09430eb1b0a9" />

Figure 3. Daily-mean precipitation averaged over the western US from StormCast UNet simulations. The no-nudging simulation is initialized from HRRR at 00:00 UTC on Mar 8th, 2017, and constrained by ERA5 data. The nudging simulation is identical except that boundary states are replaced by HRRR fields at each hour.

I am not certain whether this behavior reflects an issue with the simulation setup or a known limitation of the model. If possible, could you please advise whether there are example scripts or best practices for running longer simulations using using a local checkpoint?

Separately, I would like to mention that the training of the StormCast diffusion model is currently underway.

Thank you very much for your time and guidance!
Sincerely,
Ziming

**Comment by pzharrington** [comment]
Hi, thanks for reporting this. I frankly would not expect the model to be useful beyond 24hrs lead time or so; the predictability of km-scale features drops off quickly so beyond ~12hrs the best you can really do is sort of a dynamical downscaling from ERA5 (or similar 25km resolution forecasts, e.g. from IFS/GFS), as far as I understand. With the diffusion model you'd have hope of at least producing a fairly realistic distribution of trajectories, but it would not have much precise skill over a small regional domain at high resolution. The original paper does not evaluate the model beyond a lead time of 12 hours.

**Comment by Steven04453X** [comment]
Thank you for the detailed response and for reviewing the results, @pzharrington! I am currently working on the ensemble simulations using the diffusion model.
I was also surprise by the decline in predictability, especially given that the synoptic states (ERA5) were updated hourly. We will re-evaluate our simulation strategies, considering either a dynamical downscaling approach or re-initializing the model on a daily basis.

**Comment by Steven04453X** [comment]
Hi @pzharrington 
I hope you are doing well. I would like to share some recent results from our StormCast ensemble simulations.
Following your suggestions, we adjusted our simulation strategy. The simulations are now re-initialized every 24 hours and run for 72 hours. At each re-initialization, 10 ensemble members are generated using the diffusion model.
Figure 1 shows the hourly area-mean precipitation over the Pacific Northwest. After each re-initialization (colored dots), StormCast captures the temporal evolution of precipitation reasonably well, although peak precipitation is generally overestimated. For example, when initialized at 12:00 PM on 2025-12-07, StormCast reproduces the precipitation evolution over the following three days, but overestimates the peak precipitation at a lead time of 36 hours. We also evaluated the spatial precipitation patterns and found that StormCast demonstrates reasonable skills.

<img width="496" height="210" alt="Image" src="https://github.com/user-attachments/assets/04dc4603-cfe4-4dbf-b1be-cc8a097c478c" />

Figure 1. Area-mean hourly precipitation over the Pacific Northwest (42ºN~50ºN, 120ºW~130ºW). The black curve represents HRRR precipitation, while colored curves and shading indicate the ensemble mean and spread across the 10 StormCast members. Colored dots denote forecasts at the first lead hour after initialization.

The main issue we observe is under-dispersion in the ensemble: the error between the ensemble mean and HRRR is substantially larger than the inter-member spread.
To address this, we increase the re-initialized frequency from once per day to once every four hours. Figure 2 shows the corresponding results. The wet biases during peak precipitation is reduced under this configuration, although the under-dispersion issue remains.

<img width="468" height="227" alt="Image" src="https://github.com/user-attachments/assets/6479294a-f425-481f-9599-51b5384af7ea" />

Figure 2. Same as Figure 1 but with StormCast simulations re-initialized every four hours.

As a next step, we are exploring ways to further improve ensemble spread. One approach is to perturb the synoptic-scale input, ERA5, to generate additional ensemble variability, rather than relying solely on the diffusion model. Besides, the current inference script feeds back only the deterministic UNet prediction into the subsequent time steps. If I understand correctly, we could instead modify the script to feed back all ensemble members generated by the diffusion model into the next hour’s prediction, which may help improve ensemble diversity.

Thank you very much for your time and for your work on StormCast! And congratulation on the publication of the StormCast paper!

**Comment by pzharrington** [comment]
Interesting results! Thanks for the detailed analysis. I think in general the reduced ensemble dispersion is a known issue that was also observed in the original stormcast work. Certainly, forcing each ensemble member with the same synoptic conditioning (ERA5 in your case) will reduce the spread, so yes you could experiment with perturbed ERA5 or perhaps a GFS ensemble prediction there.

I would also point you towards our recent model, [StormScope](https://arxiv.org/abs/2601.17268), which drops the regression/deterministic component entirely and pivots to a DiT-based architecture. We have observed better ensemble calibration with this model, though it is forecasting observations directly as opposed to HRRR analysis. We are in the process of updating the [training recipe](https://github.com/NVIDIA/physicsnemo/tree/main/examples/weather/stormcast) to support both models, and this will then serve as a general-purpose training recipe for regional diffusion weather models. I will update this issue when that work is completed

**Comment by Steven04453X** [comment]
That makes sense. Thanks, @pzharrington ! We will experiment with perturbed ERA5 or GFS ensemble predictions next to address the dispersion issue.
Thank you for pointing us towrad the StormScope! After reviewing the preprint, it appears the domain covers the entire CONUS, including the western coast of US. Could you confirm if that is the case? If so, we would be happy to run some atmospheric river case simulations to compare StormScope's performance against StormCast and other physics-based models.

**Comment by pzharrington** [comment]
> After reviewing the preprint, it appears the domain covers the entire CONUS, including the western coast of US. Could you confirm if that is the case? 

That is correct. However note that the closest precip-related variable that StormScope currently predicts is base radar reflectivity, since it is observational data only. So you may have to apply some approximations/transformations on the output to get total precip.

**Comment by Steven04453X** [comment]
That sounds like a reasonable approach. In addition to precipitation and radar reflectivity, I also plan to evaluate the integrated vapor transportation (IVT) to better capture the atmospheric river dynamics. It is also very helpful to know that I can run these simulations using the existing model without the need for re-training. Please let me know when there are any updates!
